app/models/ems_refresh/save_inventory_networks.rb

```diff
@@ -168,12 +168,13 @@ def save_network_ports_inventory(ems, hashes, target = nil)
           h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
         end
 
-        h[:security_groups] = h.fetch_path(:security_groups).map { |x| x[:_object] }
+        h[:security_groups] = h.fetch_path(:security_groups).map { |x| x[:_object] } if h.fetch_path(:security_groups)
+        h[:source] = source
       end
 
-      save_inventory_multi(:network_ports, ems, hashes, deletes, [:ems_ref], nil, [])
+      save_inventory_multi(:network_ports, ems, hashes, deletes, [:mac_address], nil, [])
```

Comment:
We should just store the mac_address in the ems_ref column.  It may duplicate the data, but it makes it very clear that ems_ref is the unique reference we have chosen.